There is a test failure:
```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 7.604 s <<< FAILURE! - in io.quarkus.maven.it.GenerateConfigIT
[ERROR] testAddExtensionWithASingleExtension  Time elapsed: 7.604 s  <<< FAILURE!
java.lang.AssertionError
	at io.quarkus.maven.it.GenerateConfigIT.testAddExtensionWithASingleExtension(GenerateConfigIT.java:43)
```

--- a/core/runtime/src/main/java/io/quarkus/runtime/ExecutorTemplate.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/ExecutorTemplate.java
@@ -150,7 +150,13 @@ public class ExecutorTemplate {
         // run time config variables
         builder.setCorePoolSize(threadPoolConfig.coreThreads);
         builder.setMaximumPoolSize(threadPoolConfig.maxThreads.orElse(8 * cpus));
-        builder.setMaximumQueueSize(threadPoolConfig.queueSize);
+        if (threadPoolConfig.queueSize.isPresent()) {
+            if (threadPoolConfig.queueSize.getAsInt() < 0) {
+                builder.setMaximumQueueSize(Integer.MAX_VALUE);
+            } else {
+                builder.setMaximumQueueSize(threadPoolConfig.queueSize.getAsInt());
+            }
+        }
         builder.setGrowthResistance(threadPoolConfig.growthResistance);
         builder.setKeepAliveTime(threadPoolConfig.keepAliveTime);
         return builder.build();

--- a/core/runtime/src/main/java/io/quarkus/runtime/ThreadPoolConfig.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/ThreadPoolConfig.java
@@ -31,8 +31,8 @@ public class ThreadPoolConfig {
     /**
      * The queue size. For most applications this should be unbounded
      */
-    @ConfigItem(defaultValue = "0")
-    public int queueSize;
+    @ConfigItem
+    public OptionalInt queueSize;
 
     /**
      * The executor growth resistance.

--- a/devtools/maven/src/test/java/io/quarkus/maven/it/GenerateConfigIT.java
+++ b/devtools/maven/src/test/java/io/quarkus/maven/it/GenerateConfigIT.java
@@ -40,16 +40,16 @@ class GenerateConfigIT extends MojoTestBase {
         String file = loadFile("test.properties");
         Assert.assertTrue(file.contains("#quarkus.log.file.enable"));
         Assert.assertTrue(file.contains("If file logging should be enabled"));
-        Assert.assertTrue(file.contains("#quarkus.thread-pool.queue-size=0"));
-        Assert.assertTrue(file.contains("The queue size. For most applications this should be unbounded"));
+        Assert.assertTrue(file.contains("#quarkus.thread-pool.growth-resistance=0"));
+        Assert.assertTrue(file.contains("The executor growth resistance"));
 
         generateConfig("application.properties");
         //the existing file should not add properties that already exist
         file = loadFile("application.properties");
         Assert.assertTrue(file.contains("quarkus.log.file.enable=false"));
         Assert.assertFalse(file.contains("If file logging should be enabled"));
-        Assert.assertTrue(file.contains("#quarkus.thread-pool.queue-size=0"));
-        Assert.assertTrue(file.contains("The queue size. For most applications this should be unbounded"));
+        Assert.assertTrue(file.contains("#quarkus.thread-pool.growth-resistance=0"));
+        Assert.assertTrue(file.contains("The executor growth resistance"));
     }
 
     private String loadFile(String file) throws IOException {

